homeassistant/components/broadlink/config_flow.py

```diff
@@ -59,6 +60,22 @@ async def async_set_device(self, device, raise_on_progress=True):
             "host": device.host[0],
         }
 
+    async def async_step_dhcp(self, dhcp_discovery):
+        """Handle dhcp discovery."""
+        host = dhcp_discovery[IP_ADDRESS]
+        await self.async_set_unique_id(
+            format_mac(mac_address(dhcp_discovery[MAC_ADDRESS]))
+        )
+        self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+        try:
+            hello = partial(blk.discover, discover_ip_address=host)
+            device = (await self.hass.async_add_executor_job(hello))[0]
+        except Exception:  # pylint: disable=broad-except
```

Comment:
If it's an unexpected exception we usually want to log that at exception level to be able to figure it out. Expected exceptions we can handle specifically.

Reply:
I added handling for the expected exception types and logging for the unexpected.